lib/diff.js

```diff
@@ -68,7 +68,15 @@ Diff.blobToBuffer= function(
     binary_cb,
     hunk_cb,
     line_cb) {
-  var bufferLength = !buffer ? 0 : buffer.length;
+  var bufferText;
+  var bufferLength;
+  if (buffer instanceof Buffer) {
+    bufferText = buffer.toString("utf8");
```

Comment:
Yeah that's a good question. Certainly as a rule we shouldn't assume it. That said, we *do* assume it all over the place: e.g., https://github.com/nodegit/nodegit/blob/47e0d206a3b1a1f8114218aaee0fce514e9a9e98/lib/diff_line.js#L25 and anywhere that uses `toString` on a buffer.

I'd say addressing different encodings is a task for another day.